experimental/slo/slo_middleware.go

```diff
@@ -0,0 +1,47 @@
+package slo
+
+import (
+	"errors"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var duration = promauto.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Since histogram don't you need to specify buckets? Have you verified  subtracting histogram from histogram works as you imagine?

Reply:
I grabbed the code from [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/proxy/secure_socks_proxy.go#L38-L42) as it looked similar.  I'm no expert here but from my understanding [here](https://stackoverflow.com/questions/71768510/when-to-use-gauge-or-histogram-in-prometheus-in-recording-request-duration) we need a histogram, or we have to have shorter scrape times.

Reply:
@xnyo suggested subtracting the values, but that was with metrics captured from tempo which appear to be [histograms](https://grafana.com/docs/tempo/latest/configuration/grafana-agent/span-metrics/).  Not sure if he ever confirmed that would work.

Reply:
But wasn't he thinking taking the full latency of a trace and subtracting downstream span latencies and then update latency histogram, e.g. it's not prometheus doing the subtraction and you get one metric.

So my question is more, given you have two different latency histogram metrics how easy is it to reliably subtract one from the other and how does that scale/perform?

Reply:
I haven't tested if it subtracting histograms works reliably, but the way this histogram is set up should replicate what Tempo does with the metrics generator. I *think* @svennergr mentioned he was doing something similar with histograms in the ElasticSearch datasource as well, so he may have some guidance on whether this will be accurate or not